gpytorch/models/exact_prediction_strategies.py

```diff
@@ -69,10 +70,19 @@ def _exact_predictive_covar_inv_quad_form_cache(self, train_train_covar_inv_root
         Returns
             - A precomputed cache
         """
+        res = train_train_covar_inv_root
         if settings.detach_test_caches.on():
-            return train_train_covar_inv_root.detach()
-        else:
-            return train_train_covar_inv_root
+            res = res.detach()
+
+        def clear_cache_hook(module, *args, **kwargs):
```

Comment:
As this doesn't have any local scope, it might make sense to move this def outside this instance method and into the module. You can re-use that below as well.

Reply:
Done